### PR TITLE
[bug-fix] Enable predictions into known and unknown future in case of remaining NaN windows after imputation

### PR DESCRIPTION
**Pull request by judussoari**
Referring to Issue #744 

**Problem**
1. The code ran into an Error whenever the user opted to drop missing values that remained after imputation, with no extended df (i.e. no prediction into the unknown future)
2.  For predictions into the unknown future (i.e. using `m.make_future_dataframe()`) with `n_lags=0`, the resulting future dataframe contains `n_historic_predictions` valid values, and `periods` invalid (None/NaN) values. Feeding this df into `m._create_dataset()` either causes an Error (if `drop_missing=False`) or drops all the `periods` in the df (if `drop_missing=True`), which is not desired.

**Solution**
1. Similar to samples (inputs/targets) being dropped from the data, the same datapoints (identifiable through sample indices) must be dropped from the original/scaled df as well, in order to create the df with predictions yhat1, yhat2... 
2. For predicting into the unknown future, the NaN values at the end of `df_future` are excluded from the data drop process. This is enabled through a new kwarg `predict_steps` that is either equal to `n_forecasts` (case of forecast into known future) or `periods` (into unknown future)

### CODE REVIEW

**Comment by judussoari** [comment]
> Thank you @judussoari ! First quick question: Does your solution to 2 mean that `n_historic_predictions` is ignored or are you cropping it accordingly after appending the `periods` of None values?

Hey @ourownstory, `n_historic_predictions` is not ignored. Exactly like for `n_lags>0` before, the future df now consists of `n_historic_predictions` and `periods` NaN values. However, periods is now **always** (i.e. `n_lags>=0` instead of `n_lags>0` only) set to `m.n_forecasts`.

File: neuralprophet/forecaster.py [hunk 1]
```diff
@@ -1922,6 +1925,8 @@ def __handle_missing_data(self, df, freq, predicting):
         data_columns = []
         if self.max_lags > 0:
             data_columns.append("y")
+        if self.max_lags == 0 and predicting:  # case of n_lags = 0 and predicting into the known future
```
Comment (ourownstory) [inline neuralprophet/forecaster.py:1928]: drop

File: neuralprophet/forecaster.py [hunk 2]
```diff
@@ -1922,6 +1925,8 @@ def __handle_missing_data(self, df, freq, predicting):
         data_columns = []
         if self.max_lags > 0:
```
Comment (ourownstory) [inline neuralprophet/forecaster.py:1926]: n_lags

File: neuralprophet/time_dataset.py [hunk 3]
```diff
@@ -368,11 +373,10 @@ def _stride_lagged_features(df_col_name, feature_dims):
 
         inputs["events"] = events
 
-    if predict_mode:
```
Comment (ourownstory) [inline neuralprophet/time_dataset.py:371]: let's keep this

**Review by ourownstory — COMMENTED** [review]
When predicting (only) and dropping nan, only drop values related to lags, not n_forecasts

File: neuralprophet/time_dataset.py [hunk 4]
```diff
@@ -75,6 +75,10 @@ def drop_nan_after_init(self, drop_missing):
                 whether to automatically drop missing samples from the data
         """
         nan_idx = []
+        nan_at_end = 0
```
Comment (ourownstory) [inline neuralprophet/time_dataset.py:78]: let's get this as a kwarg `predict_steps` (n_forecasts or periods)

**Comment by ourownstory** [comment]
updated in #786

**Pull request closed without merging**